packer/conf/buildkite-agent/hooks/environment

```diff
@@ -52,31 +58,34 @@ if [[ -n "${BUILDKITE_SECRETS_BUCKET:-}" ]] ; then
 
   # Allow environment vars set in Buildkite to override paths
   secrets_prefix="${BUILDKITE_SECRETS_PREFIX:-$BUILDKITE_PIPELINE_SLUG}"
-  secrets_url_base="s3://${BUILDKITE_SECRETS_BUCKET}/${secrets_prefix}"
-  ssh_key_url="${secrets_url_base}/${SSH_KEY_NAME:-private_ssh_key}"
-  legacy_ssh_key_url="${secrets_url_base}/id_rsa_github"
-  shared_ssh_key_url="s3://${BUILDKITE_SECRETS_BUCKET}/${SHARED_SSH_KEY_NAME:-private_ssh_key}"
-  env_url="${secrets_url_base}/env"
-
-  if s3_exists "$ssh_key_url" ; then
-    echo "Downloading ssh key from $ssh_key_url"
-    ssh-add <(s3_download "$ssh_key_url")
-  elif s3_exists "$legacy_ssh_key_url" ; then
-    echo "Downloading ssh key from $legacy_ssh_key_url"
-    ssh-add <(s3_download "$legacy_ssh_key_url")
-  elif s3_exists "$shared_ssh_key_url" ; then
-    echo "Downloading ssh key from $shared_ssh_key_url"
-    ssh-add <(s3_download "$shared_ssh_key_url")
-  else
-    echo "No SSH keys found in s3://${BUILDKITE_SECRETS_BUCKET}"
-  fi
-  echo
 
-  if s3_exists "$env_url" ; then
-    echo "Downloading env from $env_url"
-    eval "$(s3_download $env_url)"
-    echo
-  fi
+  # Locations to check for ssh keys
+  ssh_key_paths=(
+    "$secrets_prefix/${SSH_KEY_NAME:-private_ssh_key}"
+    "$secrets_prefix/id_rsa_github"
+    "${SHARED_SSH_KEY_NAME:-private_ssh_key}"
+    "id_rsa_github"
+  )
+
+  for key in ${ssh_key_paths[*]} ; do
+    if s3_exists "${BUILDKITE_SECRETS_BUCKET}" "$key" ; then
+      echo "Downloading ssh-key $key"
+      SSH_ASKPASS="/bin/false" ssh-add <(s3_download "${BUILDKITE_SECRETS_BUCKET}" "$key")
+      break
+    fi
+  done
+
+  env_paths=(
+    "env"
+    "${secrets_prefix}/env"
+  )
+
+  for key in ${env_paths[*]} ; do
+    if s3_exists "${BUILDKITE_SECRETS_BUCKET}" "$key" ; then
+      echo "Downloading env $key"
+      eval "$(s3_download "${BUILDKITE_SECRETS_BUCKET}" "$key")"
```

Comment:
can be replaced with `source <(s3_download "${BUILDKITE_SECRETS_BUCKET}" "$key")`